src/vmm/src/lib.rs

```diff
@@ -301,6 +311,34 @@ impl VMM {
         Ok(())
     }
 
+    fn add_block_device(&mut self, cfg: &BlockConfig) -> Result<()> {
+        let mem = Arc::new(self.guest_memory.clone());
+
+        let range = MmioRange::new(MmioAddress(MMIO_MEM_START), 0x1000).unwrap();
+        let mmio_cfg = MmioConfig { range, gsi: 5 };
+
+        let mut guard = self.device_mgr.lock().unwrap();
+
+        let common = CommonArgs {
+            mem,
+            vm_fd: self.vm.vm_fd(),
+            event_mgr: &mut self.event_mgr,
+            mmio_mgr: guard.deref_mut(),
+            mmio_cfg,
+            kernel_cmdline: &mut self.kernel_cfg.cmdline,
+        };
+
+        let args = BlockArgs {
+            common,
+            file_path: PathBuf::from(&cfg.path),
+        };
+
+        // We can also hold this somewhere if we need to keep the handle for later.
+        let _b = Block::new(args).expect("failed to create block");
```

Comment:
We should return an error here and not use expect.

Reply:
sure